lib/icingadb/redisconnection.hpp

```diff
@@ -493,19 +493,17 @@ void RedisConnection::WriteRESP(AsyncWriteStream& stream, const Query& query, bo
 {
 	namespace asio = boost::asio;
 
-	asio::async_write(stream, asio::const_buffer("*", 1), yc);
-	WriteInt(stream, query.size(), yc);
-	asio::async_write(stream, asio::const_buffer("\r\n", 2), yc);
+	asio::streambuf writeBuffer;
+	std::ostream msg(&writeBuffer);
 
-	for (auto& arg : query) {
-		asio::async_write(stream, asio::const_buffer("$", 1), yc);
-
-		WriteInt(stream, arg.GetLength(), yc);
+	msg << "*" << query.size() << "\r\n";
 
-		asio::async_write(stream, asio::const_buffer("\r\n", 2), yc);
-		asio::async_write(stream, asio::const_buffer(arg.CStr(), arg.GetLength()), yc);
-		asio::async_write(stream, asio::const_buffer("\r\n", 2), yc);
+	for (auto& arg : query) {
+		msg << "$" << arg.GetLength() << "\r\n" << arg << "\r\n";
 	}
+
+	asio::async_write(stream, writeBuffer, yc);
+	msg.flush();
```

Comment:
Shouldn't this first flush the stream and then send the underlying buffer?

Reply:
Yep, it's kinda useless this way. But after thinking about it a bit more, why even flush at all? If I'm understanding the purpose of flush() correctly, it should sync the buffer with its output. But we don't even have an output. I'm not seeing any reason for flushing at all. Please correct me, if I'm wrong here.

Reply:
As I said: it's "just to be sure".

Reply:
https://www.boost.org/doc/libs/1_75_0/doc/html/boost_asio/reference/streambuf.html#boost_asio.reference.streambuf.examples also doesn't flush explicitly, so it should be fine.